src/platforms/react-native/sourcemaps/index.mdx

```diff
@@ -6,38 +6,299 @@ redirect_from:
+
+const config = {
+  serializer: {
+    customSerializer: createSentryMetroSerializer(),
+  },
+};
+
+const m = mergeConfig(getDefaultConfig(__dirname), config);
+module.exports = m;
+```
+
+Generate the React Native Packager (Metro) bundle and source maps:
+
+```bash {tabTitle:Android}
+npx react-native bundle \
+  --dev false \
+  --minify true \
+  --platform android \
+  --entry-file index.js \
+  --reset-cache \
+  --bundle-output index.android.bundle \
+  --sourcemap-output index.android.bundle.map
 ```
 
-If you've exported the environment variables and they match the given `release` and `dist` during the `Sentry.init` call, you can skip the next steps.
+```bash {tabTitle:iOS}
+npx react-native bundle \
+  --dev false \
+  --minify true \
+  --platform ios \
+  --entry-file index.js \
+  --reset-cache \
+  --bundle-output main.jsbundle \
+  --sourcemap-output main.jsbundle.map
+```
 
-### 2. Disable the automatic source maps upload script
+Make sure `sentry-cli` is configured for your project. For that you can use environment variables:
 
-When you set a custom release or manually upload source maps, you will need to disable the automatic source map upload script:
+<OrgAuthTokenNote />
 
-- In iOS, [revert the build script "Bundle React Native code and images"](/platforms/react-native/manual-setup/manual-setup/#bundle-react-native-code-and-images).
+```bash {filename:.env}
+SENTRY_ORG=___ORG_SLUG___
+SENTRY_PROJECT=___PROJECT_SLUG___
+SENTRY_AUTH_TOKEN=___ORG_AUTH_TOKEN___
+```
+
+Upload the bundle and source map to Sentry:
 
-- In Android, [remove the gradle integration](/platforms/react-native/manual-setup/manual-setup/#enable-gradle-integration).
+```bash {tabTitle:Android}
+node_modules/@sentry/cli/bin/sentry-cli sourcemaps upload \
+  --strip-prefix /path/to/project/root \
+  index.android.bundle index.android.bundle.map
+```
+
+```bash {tabTitle:iOS}
+node_modules/@sentry/cli/bin/sentry-cli sourcemaps upload \
+  --strip-prefix /path/to/project/root \
+  main.jsbundle main.jsbundle.map
+```
+
+## Optional Release and Distribution
+
+For events sent from Sentry to correctly be attributed to a release **both the `release` and `dist` values will need to be set.** You can set these values in the call to `init`, as discussed in our [Releases & Health content](/platforms/react-native/configuration/releases/).
+
+```js
+Sentry.init({
+  dsn: "___DSN___",
+  release: "my-project-name@2.3.12",
+  dist: "52",
+});
+```
+
+If you use custom values for your release other than the version included with the build in Xcode or Android Studio, the automatic source maps upload script will no longer work because it does not detect custom values. Use environment variables `SENTRY_RELEASE` and `SENTRY_DIST`. The script that performs the automatic source maps upload will use those values instead the default ones, and they will match the given `release` and `dist`. This ensures that the given custom `release` and `dist` values are used.
```

Comment:
```suggestion
The automatic source maps upload script doesn't detect custom values for releases. This means that if you use a custom value for your release, other than the version included with the built-in Xcode or Android Studio, the automatic source maps upload script won't work. Instead, you can set the `SENTRY_RELEASE` and `SENTRY_DIST` environment variables, which the automatic source maps upload script will use. This will ensure that the given custom `release` and `dist` values are used.
```